imagemapfunctions.js

```diff
@@ -125,23 +123,23 @@ imagemapfunctionsChecks.getCheckedElements = function (checkboxesName) {
             updatedElement = true;
         }
     }
-    if (!updatedElement){
+    if (!updatedElement) {
         console.log("No imagemap segments could be loaded from the specified checkboxes " + checkboxesName);
     }
     return checkedElements;
 }
 
-imagemapfunctionsChecks.registerImageMap = function(imageId, mapId, originalCheckboxesName, color, codeAttribute)
-{
+imagemapfunctionsChecks.registerImageMap = function (imageId, mapId, originalCheckboxesName, color, codeAttribute) {
     var checkboxesName = originalCheckboxesName;
-    var fun = function updateFunction(dictionary)
-    {
+    var fun = function updateFunction(dictionary) {
         return imagemapfunctionsChecks.updateFunctionForCheckbox(dictionary, checkboxesName);
     }
-
     var aug = ImageMapAug(imageId, mapId, fun, color);
-    aug.codeAttribute = codeAttribute;
-    aug.populateAreas(imagemapfunctionsChecks.getCheckedElements(checkboxesName));
-    aug.renderMap();
+    JSMONAME.afterRender(function () {
+        $('#'+imageId+'canvas').remove();
```

Comment:
Could you explain why this line is needed? I don't remember that this was required in the previous version, and I'd just like to understand why it is necessary.

Reply:
> Could I remove the .gitattributes, .gitignore & EM.code-workspace files after approving the PR? If I am not mistaken, these files contain some details specific to the configuration of your IDE? Correct me if I am mistaken, I am still a relative newbie with regards to GitHub.

Yes, but they don't hurt, either, as the (content of the) `.gitattributes` file will ensure that releases won't include any of the files mentioned. Do as you like.

Reply:
> Could you explain why this line is needed? I don't remember that this was required in the previous version, and I'd just like to understand why it is necessary.

Since the JS code is inlined, I figured a good way to bring in the JavascriptModuleObject was through replacement. "JSMONAME" is just a placeholder. Your approach, i.e. bringing it in via assignment to `module`, works too, but in this case has the drawback of introducing a variable in the global scope. This should be avoided.

If you are referring to the `remove()` line ... the canvas will be lost at the proper place after MLM renders the page (it still exists, though), and thus any future clicks/updates will not register with the overlay library used. Thus, this forces the canvas to be recreated.